File: Semestrul 1/Fundamentele Programarii/extra/Magazin de biciclete/service/service.py
class Service:

    def __init__(self, repository):
        self.__repository = repository

    def get_all(self):
        return self.__repository.get_all()

    def sterge_tip(self, tip):
        for el in self.get_all():
            if el.get_tip() == tip:
                self.__repository.delete(el.get_id_bicicleta())

    def sterge_max(self):
        produse = self.get_all()
        produse.sort(key = lambda x:x.get_pret(), reverse = True)
        max = produse[0].get_pret()
        for el in produse:
            if el.get_pret() == max:
                self.__repository.delete(el.get_id_bicicleta())

File: Semestrul 2/Programare orientata obiect/Pregatire_Sesiune/App/service.cpp
#pragma once

#include <algorithm>
#include "service.h"

void Service::adaugare(int id, string titlu, string tip, double pret) {
    repository.adaugare(Produs(id, titlu, tip, pret));
}

vector<Produs> Service::sortare(vector<Produs> produse) {
    sort(produse.begin(), produse.end(), [this](Produs &p1, Produs &p2) {
        return p1.get_pret() < p2.get_pret();
    });
    return produse;
}
